src/index.tsx

```diff
@@ -72,12 +72,16 @@ const smallScreenMediaQuery = window.matchMedia('(max-width: 44rem)')
 const mapActionReceiver = new MapActionReceiver(getMap(), routeStore, () => smallScreenMediaQuery.matches)
 Dispatcher.register(mapActionReceiver)
 
-getApi().infoWithDispatch() // get infos about the api as soon as possible
-
-// hook up the navbar to the query store and vice versa
 const navBar = new NavBar(getQueryStore(), getMapOptionsStore())
-// parse the initial url
-navBar.parseUrlAndReplaceQuery()
+
+// get infos about the api as soon as possible
+getApi()
+    .info()
+    .then(info => {
+        Dispatcher.dispatch(new InfoReceived(info))
+        navBar.updateStateFromUrl().then(() => navBar.startSyncingUrlWithAppState())
```

Comment:
Since we are waiting for /info here we can be sure that the /info response does not trigger a second url update (and pushes another history entry). We also wait for `updateStateFromUrl` to return to make sure we do not get multiple history entries before and after the geocoding returns.